### PR TITLE
New files engine: Prowlarr

### PR DESCRIPTION
**Pull request by narcoticfresh**
## What does this PR do?

Gives the user the possibility to search their own prowlarr instances.

Info: https://wiki.servarr.com/en/prowlarr
Github: https://github.com/Prowlarr/Prowlarr

## Why is this change important?

Prowlarr searchs multiple upstream search providers, thus allows to use that functionality through searx.

## How to test this PR locally?

* Run sudo make run
* Go to /preferences
* Enable prowlarr search engine
* Make some search
* Assert that it works as expected

## Author's checklist

This uses the prowlarr search API; it does not support pagination.

## Related issues

(none)

### CODE REVIEW

File: searx/settings.yml [hunk 1]
```diff
@@ -908,6 +908,26 @@ engines:
     engine : openstreetmap
     shortcut : osm
 
+  - name : prowlarr
```
Comment (kvch) [inline searx/settings.yml]: Could you please comment out the engine? If I understood correctly it requires config from instance admins.

File: searx/engines/prowlarr.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+"""
+
+from json import loads
+from urllib.parse import urlencode
+from searx.exceptions import SearxEngineAPIException
+
+categories = ['files']
+
+paging = False
+api_key = 'None'
```
Comment (kvch) [inline searx/engines/prowlarr.py]: Why not set it to empty string?

**Review by kvch — COMMENTED** [review]
Minor comments

File: searx/engines/prowlarr.py (hunk 2 above)
Comment (narcoticfresh) [reply searx/engines/prowlarr.py]: @kvch 
ok, addressed your comments..

i wasn't quite sure what the correct approach for the "default values" are - every engine seems a bit different there.. but i changed it to empty strings now..

**Comment by narcoticfresh** [comment]
@kvch i closed this in favor of #3118 as I had several unintended commits in this branch and I didn't want to poison the git history nor did i want to force push.. hope #3118 is alright now..

**Pull request closed without merging**